### ISSUE TITLE
Help users avoid overlapping overlays

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#1838]
### What is the problem this feature will solve?

Many users end up displaying one of the builtin overlays on top of one of the builtin base map. As a result, 
- The icons and names in the base map are covered by lines from the overlay. 
- If the overlay is based on a different map than the base map, the result has a mixture of legend from both maps.

We understand that this situation is a problematic UX, and a warning is currently shown when the overlay is turned on while a builtin base map is used. However
- The warning is not shown in the app
- The warning is not shown when opening the website
- This situation can be reached without any warning:
   1. Choose the satellite imagery as base map
   2. Show a builtin overlay on top of it
   3. Choose a builtin base map
- This situation can be reached if a share with an active overlay is opened.
- This situation is sticky: once an overlay is overlapping, the user map is distorted without any reminder of the issue.

### What is the feature you are proposing to solve the problem?

The existence of an overlay overlap will be checked
- Whenever a change is made in the layer visibility
- When the app or website is opened
- When a share is opened

If an overlap exists, the user will be asked to confirm hiding it. The confirmation will be visible on the the website and the app in both landscape and portrait orientations.

### What alternatives have you considered or tried?

1. Instead, the check could only display the existing warning. This will be less effective in reducing the issue as it will require the user to know how to hide the problematic overlay/s and to perform several clicks/touches to do it.
2. Instead, the check can avoided when opening the app/website. This will be less effective in reducing the issue because once the initial warning has been ignored, the issue persists.
3. Instead, the check can avoided when opening a share. This will be less effective in reducing the propagation of the issue through shares.
4. Keep the current functionality.

**Comment by HarelM** [comment]
The warning is shown in the app.
When opening a cloud save there is a toast, so this other toast is tricky to do.
I'm planning only to show the warning on every change in the layers in case of a problem.
I do not plan to ask the user anything, there are far better questions to ask which we have decided not to ask when opening a could save, I don't think this issue is important enough to change the decision we made.
